replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/persistence/ReplicationPersistentStore.java

```diff
@@ -82,12 +82,20 @@ public ReplicationPersistentStore(PersistentStore persistentStore) {
             String.format("Found two objects in persistence of type %s with ID %s", typeName, id));
       }
 
-      T persistable = createPersistable(klass);
-      persistable.fromMap(PersistentItem.stripSuffixes(storedMaps.get(0)));
-      return persistable;
+      T result = getPersistableFromMap(klass, storedMaps.get(0));
+      addModifiedTimeIfMissing(result);
```

Comment:
Is it more readable if I do `result = addModifiedTimeIfMissing(result);`? or better the way it is?

Reply:
✏️ I would have moved that logic to the `fromMap()` above. It is better, in my opinion, to handle all version incompatibility stuff inside to/from map methods and let the rest work with an object upgraded at the latest version